src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -266,25 +302,37 @@ case class WdlNamespace(ast: Ast, source: WdlSource, importResolver: ImportResol
    */
   def coerceRawInputs(rawInputs: WorkflowRawInputs): Try[WorkflowCoercedInputs] = {
 
-    def coerceRawInput(fqn: FullyQualifiedName, wdlType: WdlType): Try[WdlValue] = fqn match {
-      case _ if rawInputs.contains(fqn) =>
-        val rawInput = rawInputs.get(fqn).get
-        wdlType.coerceRawValue(rawInput).recoverWith {
-          case e => Failure(new UnsatisfiedInputsException(s"Failed to coerce input $fqn value $rawInput of ${rawInput.getClass} to $wdlType."))
+    def coerceRawInput(input: WorkflowInput): Try[Option[WdlValue]] = input.fqn match {
+      case _ if rawInputs.contains(input.fqn) =>
+        /* Workflow inputs can sometimes have multiple types.  For example, `${sep="," var+}`
+         * can accept a `String` or `Array[String]`.  This coercion tries to coerce to ALL valid types
+         * and just chooses the first one that works, or returns a Failure.
+         */
```

Comment:
So if I understand correctly, your suggestion is to make this a coercion responsibility.  In that case, `${sep=" " File in+}` would be stored as a `CommandParameterPart` with a wdlType of `Array[File]` in which a `File` could be coerced into an `Array[File]` for that type?

If I'm understanding correctly, I think that's a pretty good solution.